lib/Dialect/BaseModelica/Transforms/CallCSE.cpp

```diff
@@ -17,80 +19,103 @@ class CallCSEPass final : public impl::CallCSEPassBase<CallCSEPass> {
 
 private:
   mlir::LogicalResult processModelOp(ModelOp modelOp);
-
-  /// Replace all calls in the equivalence group with gets to a generated
-  /// variable. The variable will be driven by an equation derived from the
-  /// first call in the group.
-  ///
-  /// One variable and driver equation will be emitted per result,
-  /// if the call is to a function with multiple result values.
-  void emitCse(llvm::SmallVectorImpl<CallOp> &equivalenceGroup, ModelOp modelOp,
-               DynamicOp dynamicOp, mlir::SymbolTable &symbolTable,
-               mlir::RewriterBase &rewriter);
 };
 
-/// Get all call operations in the model.
-void collectCallOps(ModelOp modelOp, llvm::SmallVectorImpl<CallOp> &callOps) {
+bool haveSameInductionRanges(
+    llvm::SmallVectorImpl<EquationInstanceOp> &instanceOps) {
+  std::optional<IndexSet> seenSpace;
+  for (EquationInstanceOp instanceOp : instanceOps) {
```

Comment:
Check if you have more than one instance. If not, there's no need for checking the iteration space.